src/database/DataTypes/Transaction.js

```diff
@@ -50,6 +46,14 @@ export class Transaction extends Realm.Object {
     return this.type === 'supplier_invoice';
   }
 
+  get isExternalSupplierInvoice() {
+    return this.isSupplierInvoice && this.otherParty && this.otherParty.isExternalSupplier;
+  }
+
+  get isInternalSupplierInvoice() {
```

Comment:
new getter

Reply:
Yea that's all my thinking was. Follows the scripting pattern of calling functions after they're declared too.